src/Shared/Debugging/DebugUtils.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Utilities;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Build.Shared.Debugging
+{
+    internal static class DebugUtils
+    {
+        private enum NodeMode
+        {
+            CentralNode,
+            OutOfProcNode,
+            OutOfProcTaskHostNode
+        }
+
+        static DebugUtils()
+        {
+            string environmentDebugPath = Environment.GetEnvironmentVariable("MSBUILDDEBUGPATH");
+            var debugDirectory = environmentDebugPath ?? Path.Combine(Directory.GetCurrentDirectory(), "MSBuild_Logs");
```

Comment:
@rainersigwald @cdmihai I encountered the following unhandled exception when this code was executed from within our service hub process where we invoke MSBulild programmatically - 

I had set the `MSBuildDebugEngine` environment variable (and we have our own `BinaryLogger` specified programmatically. But I had not set `MSBuildDebugPath`.
```
System.UnauthorizedAccessException: Access to the path 'C:\Program Files\Microsoft Visual Studio\2022\Main\Common7\ServiceHub\Hosts\ServiceHub.Host.CLR.AnyCPU\MSBuild_Logs' is denied. HResult='-2147024891' 
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.Directory.InternalCreateDirectory(String fullPath, String path, Object dirSecurityObj, Boolean checkHost)
   at System.IO.Directory.InternalCreateDirectoryHelper(String path, Boolean checkHost)
   at Microsoft.Build.Shared.FileUtilities.EnsureDirectoryExists(String directoryPath)
   at Microsoft.Build.Shared.Debugging.DebugUtils..cctor()
---

System.TypeInitializationException: The type initializer for 'Microsoft.Build.Shared.Debugging.DebugUtils' threw an exception. HResult='-2146233036' 
   at Microsoft.Build.Internal.CommunicationsUtilities.GetEnvironmentVariables()
   at Microsoft.Build.Internal.Utilities.GetEnvironmentProperties()
   at Microsoft.Build.Evaluation.ProjectCollection.get_EnvironmentProperties()
   at Microsoft.Build.Evaluation.ProjectCollection.InitializeToolsetCollection(ToolsetRegistryReader registryReader, ToolsetConfigurationReader configReader)
   at Microsoft.Build.Evaluation.ProjectCollection..ctor(IDictionary`2 globalProperties, IEnumerable`1 loggers, IEnumerable`1 remoteLoggers, ToolsetDefinitionLocations toolsetDefinitionLocations, Int32 maxNodeCount, Boolean onlyLogCriticalEvents, Boolean loadProjectsReadOnly)
   at Microsoft.Build.Evaluation.ProjectCollection.get_GlobalProjectCollection()
```

Not a huge issue but now that I know about `MSBuildDebugPath` :) But have you considered moving the default location to something like `%temp%` at least for programmatic invocations so that it is harder to run into problems like above?

Reply:
Temp dir makes it a worse experience when debugging, as we're usually cd-ed into some repo directory when setting MSBuildDebugEngine and it's nice to just get all the logs there. Rather than forcing everybody to look into tmp, or forcing everybody to set MSBuildDebugPath, maybe the option that maximizes global experience is to only have users that do not propagate the user's CWD set MSBuildDebugPath.

That being said, wouldn't it be "more correct" to propagate the user's CWD to the service hub process?